presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraTokenSplitManager.java

```diff
@@ -116,8 +118,11 @@ public CassandraTokenSplitManager(CassandraSession session, int splitSize)
         return result.build();
     }
 
-    private long getTotalPartitionsCount(String keyspace, String table)
+    public long getTotalPartitionsCount(String keyspace, String table)
     {
+        if (splitsPerNode != null) {
+            return splitsPerNode;
```

Comment:
I would make it configurable via the session property. Splits per node is something very table specific. You don't want to generate 10_000 per node for a table with just a few values. And vice versa.